google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryWriteManualClientTest.java

```diff
@@ -1541,4 +1543,100 @@ public void testMultiplexingMixedLocation()
     assertEquals("us", streamWriter2.getLocation());
     assertEquals("eu", streamWriter3.getLocation());
   }
+
+  @Test
+  public void testLargeRequest() throws IOException, InterruptedException, ExecutionException {
+    String tableName = "largeRequestTable";
+    TableId tableId = TableId.of(DATASET, tableName);
+    Field col1 = Field.newBuilder("col1", StandardSQLTypeName.STRING).build();
+    Schema originalSchema = Schema.of(col1);
+    TableInfo tableInfo =
+        TableInfo.newBuilder(tableId, StandardTableDefinition.of(originalSchema)).build();
+    bigquery.create(tableInfo);
+    TableName parent = TableName.of(ServiceOptions.getDefaultProjectId(), DATASET, tableName);
+    try (StreamWriter streamWriter =
+        StreamWriter.newBuilder(parent.toString() + "/_default")
+            .setWriterSchema(
+                ProtoSchema.newBuilder()
+                    .setProtoDescriptor(
+                        DescriptorProto.newBuilder()
+                            .setName("testProto")
+                            .addField(
+                                FieldDescriptorProto.newBuilder()
+                                    .setName("col1")
+                                    .setNumber(1)
+                                    .setType(FieldDescriptorProto.Type.TYPE_STRING)
+                                    .build())
+                            .build())
+                    .build())
+            .enableLargerRequest()
+            .build()) {
+      List<Integer> sizeSet = Arrays.asList(200, 10 * 1024 * 1024, 19 * 1024 * 1024);
+      List<ApiFuture<AppendRowsResponse>> responseList =
+          new ArrayList<ApiFuture<AppendRowsResponse>>();
+      for (int i = 0; i < 10; i++) {
+        int size = sizeSet.get(new Random().nextInt(3));
+        LOG.info("Sending request of size: " + size);
+        responseList.add(
+            streamWriter.append(
+                CreateProtoRows(new String[] {new String(new char[size]).replace("\0", "a")})));
```

Comment:
can we use a more random string. We had experience on some compression will work along the way, we can eliminate that effect by random